src/transformers/integrations/integration_utils.py

```diff
@@ -197,6 +197,10 @@ def _objective(trial, checkpoint_dir=None):
                 trainer.train(resume_from_checkpoint=checkpoint)
             else:
                 trainer.train(resume_from_checkpoint=checkpoint, trial=trial)
+            # If we want to load the best model, override the object with best
+            # metric objective.
+            if use_best_model:
+                trainer.objective = trainer.compute_objective(trainer.state.best_metrics.copy())
```

Comment:
I don't like to change the internal state `trainer.objective` here. But maybe it's necessary (see the next comment) and also it's fine to change?